src/definition-builder.ts

```diff
@@ -100,6 +109,20 @@ export class DefinitionBuilder {
      * Recursively pull parameters from datum using definition template.
      */
     private extractParameters(definedDefinition: DefinitionField, templateDefinition: DefinitionField, foundParameters: DatumParameters = {}): DatumParameters {
+        if (templateDefinition instanceof Function) {
```

Comment:
Adds support for datum mappings to include dynamic pulling/pushing of parameters. One example would be an order datum not including a stake key, which will result in a different datum. Not all DEXs support this atm, but will push a fix after this PR is in